### PR TITLE
Fix compilation with newer flatbuffers

### PR DESCRIPTION
**Pull request by reshilkin**
In flatbuffers@v23.5.9 was broken forward declaration for FlatBufferBuilder. Trying to compile onnxruntime falls with the following error:
```
flatbuffers/include/flatbuffers/flatbuffer_builder.h:1420:38: error: typedef redefinition with different types ('FlatBufferBuilderImpl<false>' vs 'flatbuffers::FlatBufferBuilder')
typedef FlatBufferBuilderImpl<false> FlatBufferBuilder;
                                     ^
onnx_runtime/include/onnxruntime/core/graph/graph.h:47:11: note: previous definition is here
    class FlatBufferBuilder;
```
This PR removes these declarations and puts includes instead

### CODE REVIEW

**Comment by reshilkin** [comment]
@microsoft-github-policy-service agree

File: include/onnxruntime/core/graph/graph.h [hunk 1]
```diff
@@ -43,11 +43,7 @@
 #include "core/graph/node_arg.h"
 #include "core/graph/ort_format_load_options.h"
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
+#include "flatbuffers/flatbuffers.h"
```
Comment (edgchen1) [inline include/onnxruntime/core/graph/graph.h:46]: From here: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

"Include headers in the following order: Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers."

Please put `#include "flatbuffers/flatbuffers.h"` before the ORT header includes.

Comment (reshilkin) [reply include/onnxruntime/core/graph/graph.h]: Fixed

**Comment by edgchen1** [comment]
There are some CI build errors. E.g.:

```
2023-08-21T22:22:03.1847210Z [ 58%] Building CXX object CMakeFiles/onnxruntime_providers_xnnpack.dir/Users/runner/work/1/s/onnxruntime/core/providers/shared/node_unit/node_unit.cc.o
2023-08-21T22:22:06.7469030Z In file included from /Users/runner/work/1/s/onnxruntime/core/providers/shared/node_unit/node_unit.cc:4:
2023-08-21T22:22:06.7772670Z In file included from /Users/runner/work/1/s/onnxruntime/core/providers/shared/node_unit/node_unit.h:11:
2023-08-21T22:22:06.8221730Z /Users/runner/work/1/s/include/onnxruntime/core/graph/graph.h:23:10: fatal error: 'flatbuffers/flatbuffers.h' file not found
2023-08-21T22:22:06.8332760Z #include "flatbuffers/flatbuffers.h"
2023-08-21T22:22:06.9242470Z          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2023-08-21T22:22:08.0117840Z 1 error generated.
2023-08-21T22:22:08.0267360Z make[2]: *** [CMakeFiles/onnxruntime_providers_xnnpack.dir/Users/runner/work/1/s/onnxruntime/core/providers/shared/node_unit/node_unit.cc.o] Error 1
```

I think this is with the build flag `--use_xnnpack`.

**Comment by reshilkin** [comment]
@edgchen1
It should be better now

**Review by edgchen1 — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**